src/racer/ast.rs

```diff
@@ -896,10 +896,12 @@ pub struct TypeVisitor<'s> {
 
 impl<'ast, 's> visit::Visitor<'ast> for TypeVisitor<'s> {
     fn visit_item(&mut self, item: &ast::Item) {
-        if let ItemKind::TyAlias(_, _, _, Some(ref ty)) = item.kind {
-            self.name = Some(item.ident.name.to_string());
-            self.type_ = Ty::from_ast(&ty, self.scope);
-            debug!("typevisitor type is {:?}", self.type_);
+        if let ItemKind::TyAlias(ref ty_kind) = item.kind {
+            if let Some(ref ty) = ty_kind.3 {
+                self.name = Some(item.ident.name.to_string());
+                self.type_ = Ty::from_ast(&ty, self.scope);
+                debug!("typevisitor type is {:?}", self.type_);
+            }
```

Comment:
rustc is going to box itemkind variants to reduce the memory footprint, and the updated rustc-ap crate versions pull in some of changes that have started down that path.

if you'd prefer, we could enable the `box_patterns` feature gate for some cleaner matching, but I elected to avoid doing so in this PR